components/organisms/ContributorProfileTab/contributor-profile-tab.tsx

```diff
@@ -44,19 +55,56 @@ const ContributorProfileTab = ({
   chart,
   githubName,
   recentContributionCount,
-  repoList
+  repoList,
 }: ContributorProfileTabProps): JSX.Element => {
-  const { login } = contributor || {};
+  const { login, interests: userInterests } = contributor || {};
   const { user } = useSupabaseAuth();
+  const [selectedInterest, setSelectedInterest] = useState<string[]>([]);
+  const [showInterests, setShowInterests] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [recommendedrepos, setRecommendedRepos] = useState<string[]>([]);
+
+  const { toast } = useToast();
 
+  const interests = userInterests ? userInterests.split(",") : [];
   const { data: highlights, isError, isLoading, mutate, meta, setPage } = useFetchUserHighlights(login || "");
   const { data: emojis } = useFetchAllEmojis();
 
   const [inputVisible, setInputVisible] = useState(false);
   const pathnameRef = useRef<string | null>();
-
+  const interestArray = getInterestOptions();
   const router = useRouter();
 
+  const handleSelectInterest = (interest: string) => {
+    if (selectedInterest.length > 0 && selectedInterest.includes(interest)) {
+      setSelectedInterest((prev) => prev.filter((item) => item !== interest));
+    } else {
+      setSelectedInterest((prev) => [...prev, interest]);
+    }
+  };
+
+  const handleUpdateInterest = async () => {
+    setLoading(true);
+    const data = await updateUser({
+      data: { interests: selectedInterest },
+      params: "interests",
+    });
+    setLoading(false);
+    if (data) {
+      toast({ description: "Updated successfully", variant: "success" });
+    } else {
+      toast({ description: "An error occured!", variant: "danger" });
+    }
+  };
+
+  const getRepoFullNameByInterests = () => {
+    const repoFullNames = interests.map((interest) => {
+      return recommendations[interest as interestsType].toString();
```

Comment:
Why use `toString` here? This should be able to handle multiple repos per interest for the future